File: seed/csharp-sdk/trace/src/SeedTrace/Submission/Types/TestCaseResultWithStdout.cs
using System.Text.Json.Serialization;
using SeedTrace.Core;

namespace SeedTrace;

public record TestCaseResultWithStdout
{
    [JsonPropertyName("result")]
    public required TestCaseResult Result { get; set; }

    [JsonPropertyName("stdout")]
    public required string Stdout { get; set; }

    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/rust-sdk/trace/src/types/test_case_result_with_stdout.rs
use crate::test_case_result::TestCaseResult;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct TestCaseResultWithStdout {
    pub result: TestCaseResult,
    pub stdout: String,
}